### PR TITLE
Declare the inputs/outputs properly so the compile tasks are incremental

### PR DESCRIPTION
**Pull request by AlecKazakova**

### CODE REVIEW

File: gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
+  @InputFiles lateinit var inputs: FileCollection
+
+  @OutputDirectories lateinit var outputDirectories: FileCollection
+
+  @Input internal var buildType: NativeBuildType? = null
+    set(value) {
+      val outputs = mutableListOf("${project.buildDir.path}/${project.name}.framework")
+      if (hasDsyms()) outputs.add("${project.buildDir.path}/${project.name}.framework.dSYM")
+      outputDirectories = project.files(*outputs.toTypedArray())
+      field = value
```
Comment (jrodbx) [inline gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt:22]: Setting this value has a side effect of updating the outputDirectories.  In doing so, _I think_ that call to hasDsyms() is going to resolve that outputFile (via get()) -- an operation that should typically be confined to the task action.  Any way that side effect logic can be moved out of this setter and into the task action?

File: gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt [hunk 2]
```diff
@@ -27,14 +42,21 @@ open class CocoapodsCompileTask : DefaultTask() {
...
-              bundleName = "${project.name}.framework.dSYM"
+          binaryPath = "Contents/Resources/DWARF/${project.name}",
+          bundleName = "${project.name}.framework.dSYM"
       )
     }
   }
 
+  private fun hasDsyms(): Boolean {
+    return compilations.all { compilation ->
+      val output = compilation.outputFile.get().parentFile.absolutePath
```
Comment (jrodbx) [inline gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt]: this `get()` is what i'm referring to above.

File: gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
-import org.jetbrains.kotlin.konan.target.KonanTarget.IOS_ARM64
-import org.jetbrains.kotlin.konan.target.KonanTarget.IOS_X64
 import java.io.File
 
 open class CocoapodsCompileTask : DefaultTask() {
-  internal lateinit var buildType: NativeBuildType
-  internal lateinit var compilations: Collection<KotlinNativeLink>
+  @InputFiles lateinit var inputs: FileCollection
+
+  @OutputDirectories lateinit var outputDirectories: FileCollection
```
Comment (jrodbx) [inline gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt]: nit: move `@OutputDirectories` out from in between @Input...s

**Review by jrodbx — APPROVED** [review]

File: gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt (hunk 1 above)
Comment (AlecKazakova) [reply gradle-plugin/src/main/kotlin/com/alecstrong/cocoapods/gradle/plugin/CocoapodsCompileTask.kt:22]: totally. You also brought attention to a mistake I made where its actually checking for the physical output file of a task that hasnt run yet - I have a fix

**Pull request merged**